skimage/util/_flood_fill.py

```diff
@@ -0,0 +1,180 @@
+        Adjacent pixels whose squared distance from the center is larger or
+        equal to `connectivity` are considered neighbors. Ignored if
+        `selem` is not None.
+    indices : bool, optional
+        If True, the output will be an array representing indices of the flood
+        fill. If False (default), the output will be an n-dimensional array
+        with flood filling applied.
+    tolerance : ``selem`` type, optional
+        If None (default), adjacent values must be strictly equal to the
+        initial value of `image` at `seed_point`.  This is fastest.  If a value
+        is given, a comparison will be done at every point and this tolerance
+        on each side of the initial value will also be filled (inclusive).
+    inplace : bool, optional
+        If True, flood filling is applied to `image` inplace.  If False, the
+        flood filled result is returned without modifying the input `image`
+        (default).  Ignored if `indices` is True.
+
+    Returns
+    -------
+    filled : ndarray or tuple[ndarray]
+        If `indices` is false, an array with the same shape as `image` is
+        returned with values equal to (or within tolerance of) the seed point
+        set to `new_value`.  If `indices` is true, a tuple of one-dimensional
+        arrays containing the coordinates (indices) of all found maxima is
+        returned.
+
+    Notes
+    -----
+    The conceptual analogy of this operation is the 'paint bucket' tool in many
+    raster graphics programs.
+
+    Examples
+    --------
+    >>> from skimage.util import flood_fill
+    >>> image = np.zeros((4, 7), dtype=int)
+    >>> image[1:3, 1:3] = 1
+    >>> image[3, 0] = 1
+    >>> image[1:3, 4:6] = 2
+    >>> image[3, 6] = 3
+    >>> image
+    array([[0, 0, 0, 0, 0, 0, 0],
+           [0, 1, 1, 0, 2, 2, 0],
+           [0, 1, 1, 0, 2, 2, 0],
+           [1, 0, 0, 0, 0, 0, 3]])
+
+    Fill connected ones with 5, with full connectivity (diagonals included):
+
+    >>> flood_fill(image, (1, 1), 5)
+    array([[0, 0, 0, 0, 0, 0, 0],
+           [0, 5, 5, 0, 2, 2, 0],
+           [0, 5, 5, 0, 2, 2, 0],
+           [5, 0, 0, 0, 0, 0, 3]])
+
+    Fill connected ones with 5, with only cardinal direction connectivity:
+
+    >>> flood_fill(image, (1, 1), 5, connectivity=1)
+    array([[0, 0, 0, 0, 0, 0, 0],
+           [0, 5, 5, 0, 2, 2, 0],
+           [0, 5, 5, 0, 2, 2, 0],
+           [1, 0, 0, 0, 0, 0, 3]])
+
+    Fill with a tolerance:
+
+    >>> flood_fill(image, (0, 0), 5, tolerance=1)
+    array([[5, 5, 5, 5, 5, 5, 5],
+           [5, 5, 5, 5, 2, 2, 5],
+           [5, 5, 5, 5, 2, 2, 5],
+           [5, 5, 5, 5, 5, 5, 3]])
+    """
+    from ..morphology.extrema import (_resolve_neighborhood,
+                                      _set_edge_values_inplace, _fast_pad)
+    from ..morphology.watershed import _offsets_to_raveled_neighbors
+    from ._flood_fill_cy import _flood_fill_equal, _flood_fill_tolerance
+
+    image = np.asarray(image, order='C')
```

Comment:
Will this clash with `inplace=True`? If `image` is not C-contiguous this will perform a copy and the original input image is lost.